### ISSUE TITLE
Pi-hole does not resolve own hostname even though present in /etc/hosts

### ISSUE THREAD

**Issue by Schwefelholz** [pi-hole/FTL#1201]
### Versions

  Pi-hole version is v5.5 (Latest: v5.5)
  AdminLTE version is v5.7 (Latest: v5.7)
  FTL version is v5.10.1 (Latest: v5.10.2)

### Platform

- OS and version:  Raspbian GNU/Linux 9 and Debian GNU/Linux 10

- Platform:  Raspberry Pi and VM

### Expected behavior

Clients should be able to resolve the hostname of servers on which pi-hole is installed

### Actual behavior / bug

I've got two installations of pi-hole on platforms/OS mantioned above. Both have recently been updated to the latest pi-hole release.
Since then, both servers do not resolve their own hostnames anymore even though these are listed in /etc/hosts

### Steps to reproduce

runget@saturn:~$ nslookup
> server 192.168.128.8
Default server: 192.168.128.8
Address: 192.168.128.8#53
> mercury
Server:		192.168.128.8
Address:	192.168.128.8#53

Non-authoritative answer:
*** Can't find mercury.egnur.priv: No answer
> proxima
Server:		192.168.128.8
Address:	192.168.128.8#53

Name:	proxima.egnur.priv
Address: 192.168.128.25
> 
> 
> server 192.168.128.25
Default server: 192.168.128.25
Address: 192.168.128.25#53
> mercury
Server:		192.168.128.25
Address:	192.168.128.25#53

Name:	mercury.egnur.priv
Address: 192.168.128.8
> 
> 
> proxima
Server:		192.168.128.25
Address:	192.168.128.25#53

Non-authoritative answer:
*** Can't find proxima.egnur.priv: No answer
> 

## Debug Token

- URL:  (mercury) https://tricorder.pi-hole.net/avatYtRW/
- URL:  (proxima) https://tricorder.pi-hole.net/0FWjkKVA/

### Additional context

/etc/hosts (excerpt, mercury)
runget@mercury:~ $ cat /etc/hosts | grep -iE "mercury|proxima"
pi-hole/pi-hole#127.0.1.1	mercury
192.168.128.8	mercury.egnur.priv	mercury		mercury.egnur.eu	hermes.egnur.eu
192.168.128.25	proxima.egnur.priv	proxima		unifi unifi.egnur.priv	lms

/etc/hosts (excerpt, proxima)
runget@proxima:~$ cat /etc/hosts | grep -iE "mercury|proxima"
pi-hole/pi-hole#127.0.1.1	proxima
192.168.128.8	mercury.egnur.priv	mercury		mercury.egnur.eu	hermes.egnur.eu
192.168.128.25	proxima.egnur.priv	proxima		unifi unifi.egnur.priv	lms

**Comment by DL6ER** [comment]
@Schwefelholz do you depend on `bind-interfaces` despite its drawbacks? If not, please remove it. If yes, please check if
```
pihole checkout ftl tweak/iface_detection
```
fixes the issue for you.

**Comment by Schwefelholz** [comment]
@DL6ER  I'm not sure if it is actually needed. I'm running unbound on the pi-hole servers as well. The setup is as of given in [this article](https://forum.kuketz-blog.de/viewtopic.php?t=3067).

**Comment by DL6ER** [comment]
Yes, I see
```
port: 5335
```
for unbound is this guide. Could you check if removing this line and restarting the Pi-hole fixes the issue for you?

**Comment by Schwefelholz** [comment]
Yes, it seems commenting out that line resolves the issue. Not sure, yet, if it raises any others... ;-)

**Comment by DL6ER** [comment]
Where did you get this line from? Why was it recommended?

**Comment by Schwefelholz** [comment]
I really do not remember. Before having implemented pi-hole, I already ran a dnsmasq setup which I tried to take over to pi-hole. The setting might have been taken over from this old setup.

A question from my end is: Why is it suddenly making problems, given it is there for quite a long time?

**Comment by DL6ER** [comment]
This change was introduced with [FTL v5.9](https://pi-hole.net/2021/09/11/pi-hole-ftl-v5-9-web-v5-6-and-core-v5-4-released/#page-content), emphasis added:

> **Interface-dependent handling of pi.hole and the machine’s hostname**
> This makes FTL automatically reply with the appropriate IP address to both pi.hole and the machines hostname. *Before this change, FTL always used a hard-coded address set during the weekly gravity updates* (pihole -g). The new method is interface-aware and may reply with different addresses on different interfaces (e.g. Ethernet, WiFi or Wireguard network). The address FTL replies with can be overwritten using the `REPLY_ADDR4/6` settings in `/etc/pihole/pihole-FTL.conf`.

If using `bind-interfaces`, this interface-awareness is tricky and `bind-interfaces` we only recently found as being able to trick this under some configurations. This is what https://github.com/pi-hole/FTL/pull/1198 is going to fix for the next release.
